app/controllers/api/v1/recordings_controller.rb

```diff
@@ -57,19 +59,27 @@ def resync
         render_json(status: :ok)
       end
 
-      def publish_recording
-        publish = params[:publish]
-        record_id = params[:record_id]
-        case publish
-        when 'true'
-          visibility = 'Published'
-        when 'false'
-          visibility = 'Unpublished'
-        end
+      # POST /api/v1/recordings/update_visibility.json
+      def update_visibility
+        record_id = params[:id]
+        visibility = params[:visibility]
+
+        # Updates the recording locally
         Recording.find_by(record_id:).update(visibility:)
```

Comment:
Shouldn't we update the record on the DB after we update the recording data on the BBB?
This is safer for cases where the requests to BBB fails i.e. a network issue we won't update the recording state unless it's updated on the BBB server.
Plus on the `send_XXX_request` I think we count on checking the old and new visibility values that will always be the same since we're calling them after updating the recrod.
That means that both methods will always be turthy and therefore the optmisation isn't happening.